File: src/main/java/fr/lewon/web/bot/entities/input/battle/ActionBattle.java
package fr.lewon.web.bot.entities.input.battle;

import java.math.BigDecimal;

import fr.lewon.bot.http.body.urlencoded.FUEMember;
import fr.lewon.web.bot.entities.battle.AbstractBattleOpponent;
import fr.lewon.web.bot.entities.input.Action;

public abstract class ActionBattle extends Action {

	@FUEMember("who[orgasm]")
	private Integer whoOrgasm;

	@FUEMember("who[ego]")
	private BigDecimal whoEgo;

	@FUEMember("who[x]")
	private Integer whoX;

	@FUEMember("who[curr_ego]")
	private BigDecimal whoCurrEgo;

	@FUEMember("who[nb_org]")
	private Integer whoNbOrg;

	@FUEMember("who[figure]")
	private Integer whoFigure;

	@FUEMember
	private Integer autoFight;


	public ActionBattle() {
		this(null);
	}

	public ActionBattle(AbstractBattleOpponent opponent) {
		super("Battle", "fight");
		if (opponent != null) {
			this.whoOrgasm = opponent.getOrgasm();
			this.whoEgo = opponent.getEgo();
			this.whoX = opponent.getX();
			this.whoCurrEgo = opponent.getCurrEgo();
			this.whoNbOrg = opponent.getNbOrg();
			this.whoFigure = opponent.getFigure();
		}
		this.autoFight = 0;
	}


	public Integer getWhoOrgasm() {
		return whoOrgasm;
	}

	public void setWhoOrgasm(Integer whoOrgasm) {
		this.whoOrgasm = whoOrgasm;
	}

	public BigDecimal getWhoEgo() {
		return whoEgo;
	}

	public void setWhoEgo(BigDecimal whoEgo) {
		this.whoEgo = whoEgo;
	}

	public Integer getWhoX() {
		return whoX;
	}

	public void setWhoX(Integer whoX) {
		this.whoX = whoX;
	}

	public BigDecimal getWhoCurrEgo() {
		return whoCurrEgo;
	}

	public void setWhoCurrEgo(BigDecimal whoCurrEgo) {
		this.whoCurrEgo = whoCurrEgo;
	}

	public Integer getWhoNbOrg() {
		return whoNbOrg;
	}

	public void setWhoNbOrg(Integer whoNbOrg) {
		this.whoNbOrg = whoNbOrg;
	}

	public Integer getWhoFigure() {
		return whoFigure;
	}

	public void setWhoFigure(Integer whoFigure) {
		this.whoFigure = whoFigure;
	}

	public Integer getAutoFight() {
		return autoFight;
	}

	public void setAutoFight(Integer autoFight) {
		this.autoFight = autoFight;
	}




}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/battle/ActionBattle.kt
package fr.lewon.bot.hh.entities.input.battle

import com.fasterxml.jackson.annotation.JsonProperty
import fr.lewon.bot.hh.entities.battle.AbstractBattleOpponent
import fr.lewon.bot.hh.entities.input.Action
import java.math.BigDecimal

abstract class ActionBattle @JvmOverloads constructor(opponent: AbstractBattleOpponent? = null) : Action("Battle", "fight") {
    @JsonProperty("who[orgasm]")
    val whoOrgasm: Int? = opponent?.orgasm
    @JsonProperty("who[ego]")
    val whoEgo: BigDecimal? = opponent?.ego
    @JsonProperty("who[x]")
    val whoX: Int? = opponent?.x
    @JsonProperty("who[curr_ego]")
    val whoCurrEgo: BigDecimal? = opponent?.currEgo
    @JsonProperty("who[nb_org]")
    val whoNbOrg: Int? = opponent?.nbOrg
    @JsonProperty("who[figure]")
    val whoFigure: Int? = opponent?.figure
    @JsonProperty
    val autoFight: Int = 0
}